mlir/include/mlir/Dialect/LLVMIR/GENXOps.td

```diff
@@ -148,14 +148,71 @@ def GENX_BarrierOp : GENX_Op<"barrier"> {
   let assemblyFormat = "attr-dict";
 }
 
-def ShflKindXor : I32EnumAttrCase<"XOR", 0>;
-def ShflKindUp   : I32EnumAttrCase<"UP", 1>;
-def ShflKindDown : I32EnumAttrCase<"DOWN", 2>;
-def ShflKindIdx  : I32EnumAttrCase<"IDX", 3>;
+/// Enum attribute of the different memory fence flags.
+def MemoryFenceFlag : I32EnumAttr<"MemoryFenceFlag", "GENX memory fence flag",
+  [
+    I32EnumAttrCase<"CLK_LOCAL_MEM_FENCE",  1>,
```

Comment:
We should use enum names that aren't tied to OpenCL, so CLK -> GENX or remove CLK